draft-yoshino-wish-00.txt

```diff
@@ -321,6 +257,70 @@ Internet-DrafWiSH: A General Purpose Message Framing over B October 2016
    needed.  What contents are exchanged and in what encoding they are
    exchanged over WiSH are to be defined by the application layer.
 
+6.  Using WiSH over HTTP
+
+   Standard HTTP (REST) semantics should be followed, especially the
+   choice of the HTTP method.  Some HTTP semantics may not be
+   applicable, e.g.  Cache-Control, when the body is streamed.  However,
+   such limitation is not specific to WiSH.
+
+7.  WebSocket Compatibility
+
+   In this section, we discuss how to bridge WiSH and WebSocket (API and
+   protocol).
+
+7.1.  Subprotocol Negotiation
+
+   When layered over HTTP, WiSH optionally negotiates the subprotocol,
+   i.e. the protocol layered over WiSH, by using the Accept header and
+   Content-Type header of the underlying HTTP.  RFC 6455 negotiated the
+
+
+
+Yoshino & Zhu              Expires May 2, 2017                  [Page 5]
+
+Internet-DrafWiSH: A General Purpose Message Framing over B October 2016
+
+
+   subprotocol by exchanging the Sec-WebSocket-Protocol header.  WiSH
+   lists the offered subprotocols in the Accept header as follows:
+
+       Accept: application/web-stream; protocol=foo; q=1,
+           application/web-stream; protocol=bar; q=0.5
+
+   Following the "application/web-stream" media type, a parameter named
+   "protocol" can be specified with the subprotocol name as its value.
+   Multiple subprotocols are offered by listing multiple application/
+   web-stream media type with the protocol parameter with different
+   values.  The server chooses one subprotocol from the offered ones and
+   notifies the chosen subprotocol by including the Content-Type header
+   with the WiSH media type with the subprotocol as its value as
+   follows:
+
+       Content-Type: application/web-stream; protocol=foo
+
+   WiSH does not forbid use of asymmetric subprotocols as different
+   values can be used for the Content-Type header in the HTTP request
+   and HTTP response (see Section 6).  This includes WiSH and non-WiSH
+   media type combinations.  However, symmetric subprotocols MUST be
+   used where compatibility with WebSocket matters.  A client MAY use
```

Comment:
The first half of this paragraph looks good.

however, the 2nd half should be moved to before you talk about asymmetric support.